crate/utils/src/crypto/rsa/operation.rs

```diff
@@ -0,0 +1,144 @@
+    );
+
+    Object::PrivateKey {
+        key_block: KeyBlock {
+            cryptographic_algorithm: Some(CryptographicAlgorithm::DH),
+            key_format_type: KeyFormatType::TransparentRSAPrivateKey,
+            key_compression_type: None,
+            key_value: KeyValue {
+                key_material: KeyMaterial::TransparentRSAPrivateKey {
+                    modulus: BigUint::from_bytes_be(&private_key.n().to_vec()),
+                    private_exponent: Some(BigUint::from_bytes_be(&private_key.d().to_vec())),
+                    public_exponent: Some(BigUint::from_bytes_be(&private_key.e().to_vec())),
+                    p: private_key.p().map(|p| BigUint::from_bytes_be(&p.to_vec())),
+                    q: private_key.q().map(|q| BigUint::from_bytes_be(&q.to_vec())),
+                    prime_exponent_p: private_key
+                        .dmp1()
+                        .map(|dmp1| BigUint::from_bytes_be(&dmp1.to_vec())),
+                    prime_exponent_q: private_key
+                        .dmq1()
+                        .map(|dmq1| BigUint::from_bytes_be(&dmq1.to_vec())),
+                    crt_coefficient: private_key
+                        .iqmp()
+                        .map(|iqmp| BigUint::from_bytes_be(&iqmp.to_vec())),
+                },
+                attributes: Some(Attributes {
+                    object_type: Some(ObjectType::PrivateKey),
+                    cryptographic_algorithm: Some(CryptographicAlgorithm::DH),
+                    cryptographic_length: Some(cryptographic_length_in_bits),
+                    cryptographic_usage_mask: Some(CryptographicUsageMask::Decrypt),
+                    vendor_attributes: None,
+                    key_format_type: Some(KeyFormatType::TransparentRSAPrivateKey),
+                    cryptographic_parameters: Some(CryptographicParameters {
+                        cryptographic_algorithm: Some(CryptographicAlgorithm::DH),
+                        ..CryptographicParameters::default()
+                    }),
+                    cryptographic_domain_parameters: Some(CryptographicDomainParameters {
+                        q_length: Some(pkey_bits_number as i32),
+                        recommended_curve: None,
+                    }),
+                    link: Some(vec![Link {
+                        link_type: LinkType::PublicKeyLink,
+                        linked_object_identifier: LinkedObjectIdentifier::TextString(
+                            public_key_uid.to_string(),
+                        ),
+                    }]),
+                    ..Attributes::default()
+                }),
+            },
+            cryptographic_length: Some(cryptographic_length_in_bits),
+            key_wrapping_data: None,
+        },
+    }
+}
+
+pub fn create_rsa_key_pair(
+    key_size: u32,
+    public_key_uid: &str,
+    private_key_uid: &str,
+) -> Result<KeyPair, KmipUtilsError> {
+    let rsa_private = Rsa::generate(key_size)?;
```

Comment:
check key_size is > fips minimum key size for rsa
```suggestion
    #[cfg(feature = "fips")]
    if key_size < FIPS_MIN_KEY_SIZE {
        bail!()
    }
    let rsa_private = Rsa::generate(key_size)?;
```
And `FIPS_MIN_KEY_SIZE` is defined in wrap/rsa_oaep_aes_kwp.rs but you can define it here and import it there.

Reply:
thanks